src/coreclr/src/jit/emitxarch.cpp

```diff
@@ -14447,156 +14530,319 @@ emitter::insExecutionCharacteristics emitter::getInsExecutionCharacteristics(ins
+        case INS_punpckhdq:
+        case INS_punpckhwd:
+        case INS_punpckhbw:
+        case INS_punpcklqdq:
+        case INS_punpckhqdq:
+        case INS_pshufb:
+        case INS_pshufd:
+        case INS_pshuflw:
+        case INS_pshufhw:
+        case INS_shufps:
+        case INS_shufpd:
+        case INS_pblendw:
+        case INS_movsldup:
+        case INS_movshdup:
+        case INS_insertps:
+        case INS_palignr:
+        case INS_vpermilps:
+        case INS_vpermilpd:
+        case INS_vpermilpsvar:
+        case INS_vpermilpdvar:
+            result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+            result.insLatency += PERFSCORE_LATENCY_1C;
+            break;
+
+        case INS_vblendvps:
+        case INS_vblendvpd:
+        case INS_vpblendvb:
             result.insThroughput = PERFSCORE_THROUGHPUT_1C;
             if (memAccessKind == PERFSCORE_MEMORY_NONE)
             {
-                result.insLatency = PERFSCORE_LATENCY_3C;
+                result.insLatency = PERFSCORE_LATENCY_2C;
+            }
+            break;
+
+        case INS_bswap:
+            if ((id->idOpSize() == EA_8BYTE))
+            {
+                result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+                result.insLatency    = PERFSCORE_LATENCY_2C;
             }
             else
             {
-                result.insLatency = max(PERFSCORE_LATENCY_5C, result.insLatency);
+                assert(id->idOpSize() == EA_4BYTE);
+                result.insThroughput = PERFSCORE_THROUGHPUT_2X;
+                result.insLatency    = PERFSCORE_LATENCY_1C;
             }
             break;
 
-        case INS_mulx:
+        case INS_pmovmskb:
+        case INS_movmskpd:
+        case INS_movmskps:
             result.insThroughput = PERFSCORE_THROUGHPUT_1C;
-            result.insLatency    = PERFSCORE_LATENCY_4C;
+            result.insLatency += PERFSCORE_LATENCY_2C;
             break;
 
         case INS_pextrb:
         case INS_pextrd:
-        case INS_pextrq:
         case INS_pextrw:
+        case INS_pextrq:
         case INS_pextrw_sse41:
+        case INS_lzcnt:
+        case INS_tzcnt:
+        case INS_popcnt:
+        case INS_crc32:
+        case INS_pdep:
+        case INS_pext:
+        case INS_addsubps:
+        case INS_addsubpd:
+        case INS_pcmpgtq:
+        case INS_psadbw:
+        case INS_vpermps:
+        case INS_vpermpd:
+        case INS_vpermd:
+        case INS_vpermq:
+        case INS_vperm2i128:
+        case INS_vperm2f128:
+        case INS_pmovsxbw:
+        case INS_pmovsxbd:
+        case INS_pmovsxbq:
+        case INS_pmovsxwd:
+        case INS_pmovsxwq:
+        case INS_pmovsxdq:
+        case INS_pmovzxbw:
+        case INS_pmovzxbd:
+        case INS_pmovzxbq:
+        case INS_pmovzxwd:
+        case INS_pmovzxwq:
+        case INS_pmovzxdq:
+            result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+            result.insLatency += PERFSCORE_LATENCY_3C;
```

Comment:
It seems that these have unit latency for xmm registers?

Reply:
These instructions have a latency of 3 cycles according to uops.info
The Agner table omits these instructions for the Skylake-X